File: src/main/java/com/mx/imsoftware/service/PersonaService.java
package com.mx.imsoftware.service;

import com.mx.imsoftware.model.Persona;

import java.util.List;

/**
 * Interfaz que define los métodos del servicio para la gestión de personas.
 */
public interface PersonaService {
    /**
     * Guarda una nueva persona en el sistema.
     *
     * @param persona Objeto con los datos de la persona a guardar
     * @return Persona guardada
     */
    Persona savePersona(Persona persona);

    /**
     * Obtiene un listado de las personas.
     *
     * @return Listado de personas
     */
    List<Persona> findAllPersonas();
}

File: frontend/src/services/PersonaService.ts
import axios from "axios";
import { Persona } from "./../interfaces/Persona"; // ajusta el path si va en otro archivo

const BASE_REST_API_URL = import.meta.env.VITE_API_BASE_URL;

/**
 * Obtiene listado de personas registradas
 * @returns Promesa con la respuesta del backend
 */
export const getAllPersonas = () => axios.get<Persona[]>(BASE_REST_API_URL);

/**
 * Guarda una nueva persona en la base de datos
 * @param persona Objeto con nombre, email y edad
 * @returns Promesa con la respuesta del backend
 */
export const savePersona = (Persona: Omit<Persona, 'id'>) =>
  axios.post<Persona>(BASE_REST_API_URL, Persona);
